src/index.scss

```diff
@@ -0,0 +1,4 @@
+@use "./base";
+// The "as _" is necessary because otherwise
+// Sass tries to use the * from the glob as the namespace and fails
+@use './components/**/*.scss' as _;
```

Comment:
Would this be a better fit for [`@forward`](https://sass-lang.com/documentation/at-rules/forward)?

Reply:
My understanding is that `@forward` is like an `export * from "..."` in JS. Initially I didn't think that is what we want because for this file, I intended that the only target usage is for it to be generated into the css file in `dist`. But then it occurred to me that people might also want to import this file directly as a sass file. Is that a valid use case?

Reply:
I'm honestly not sure, as I'm just getting back into the Sass world. But it seems like our options are:

- `@use`, which requires a namespace to be specified (as mentioned in your comment) even though we'll never be using it
- [`@import`](https://sass-lang.com/documentation/at-rules/import), which does not require a namespace but Sass discourages its continued use
- `@forward`, which does not require a namespace and will make whatever was included available when imported and is intended to be used when splitting a single package up among multiple files

Of the three, `@forward` felt the most appropriate to me. But I might be wrong!

Reply:
I think that in this context the only difference between @use and @forward is that if another module @use'd the `index.scss` file, then that other module would have access to mixins from the components and other files. But I only created this `index.scss` file to generate the `dist` css file, it isn't intended to be `@use`'d.

Reply:
So if I am understanding this correctly (and I may not be)...

- Both techniques result in the same bundled `*.css` file.
- If we use `@use` (heh), you won't be able to access mixins, variables, etc. via Sass except through a leading underscore namespace.
- If we use `@forward`, you _would_ be able to access mixins, variables, etc. via Sass.

If that's accurate, I still think `@forward` sounds like what we want? Or am I missing somethin?

Reply:
I was planning on adding this file to the npm exclude list so it doesn't even get published. This file is not intended to be used, other than to generate the css file in dist. So there is no need to export mixins, variables, etc. from this file.

Reply:
This is for the same reason that I wouldn't export anything from an entry-point JS file. It's not going to get imported anywhere, so there is no need to export anything

Reply:
Okay, so is this more accurate then?

- Both techniques result in the same bundled `*.css` file.
- If we use `@use`, we have to specify a `_` namespace.
- If we use `@forward`, it just works as expected.

Reply:
"it just works as expected"

In this scenario, I don't want to export anything, so @forward does not do what I expect.

Reply:
> In this scenario, I don't want to export anything, so @forward does not do what I expect.

Maybe we need to hop on a call at some point, I'm very confused.

I find this line of SCSS:

```scss
@use './components/**/*.scss' as _;
```

More confusing than this one:

```scss
@forward './components/**/*.scss';
```

It sounds like you're worried about `index.scss` exporting things. But you also say that it is "not going to get imported anywhere." So I don't understand what the issue is.

Is the bundle CSS different? Are there regressions? Is there an API key in our Sass variables we don't want to expose? Do underscores give us street cred? What am I missing?

Reply:
I realized that the underscore may be confusing so I changed it to `components`: https://github.com/cloudfour/cloudfour.com-patterns/pull/569/commits/36b6c394877356b61d2617224edc3ec5548ff392#diff-5ab26e17c78a72ee4bc1774adbcf305eR4

For me, `@forward` raises more questions when I see it because it makes me think "where is this forwarding to, what is supposed to be importing this". `@use` is more clear to me because it just says what it does, it is only loading in another file.

It's like in JS using `export * from "..."` as an import rather than using `import`. I wouldn't use an export statement just for the purpose of importing. I would use an import statement instead.

Reply:
I think the majority of us interacting with the Sass won't have that same point of confusion. But just to double check, I decided to see what other large UI frameworks are doing.

- [Bootstrap is sticking with `@import` for now](https://github.com/twbs/bootstrap/issues/30025#issuecomment-574825600), as are Salesforce, Mozilla Protocol and Shopify Polaris.
- Material uses `@use` but no globs (just manually lists packages 🤔): https://github.com/material-components/material-components-web/blob/master/packages/material-components-web/material-components-web.scss

In light of that, I think I'm okay with this. I still disagree, but I no longer feel passionate about it.